tests/integration/conftest.py

```diff
@@ -0,0 +1,116 @@
+#  Copyright (c) ZenML GmbH 2022. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+import os
+import shutil
+from typing import Generator
+
+import pytest
+from pytest_mock import MockerFixture
+
+from zenml.container_registries import BaseContainerRegistry
+from zenml.repository import Repository
+from zenml.stack import Stack
+
+
+@pytest.fixture(scope="module")
+def shared_kubeflow_repo(
+    base_repo: Repository,
+    tmp_path_factory: pytest.TempPathFactory,
+    module_mocker: MockerFixture,
+) -> Generator[Repository, None, None]:
```

Comment:
Is there a way we could keep the ability to run tests without Kubeflow ? Not all OS/archs support Kubeflow and even on those that do, I see some value for running these tests locally without involving kubeflow.

Reply:
Yes I do have some ideas on how to make this distinction if we need it, e.g. pytest allows you to custom command line options (https://docs.pytest.org/en/6.2.x/example/simple.html#dynamically-adding-command-line-options). We could run our integration tests with `pytest tests/integration --with-kubeflow` or something similar to specify whether kubeflow should be included